### PR TITLE
Add try catch monitoring process starter

### PR DESCRIPTION
**Pull request by ZhuozhaoLi**
This PR is another fix mentioned in #1267 . This includes:

1. Wrapping the Hub and DBM starter with a `try catch`. If there is exception on Hub or DBM, let `MonitoringHub` know through a multiprocessing queue (any thought of potential problem on that?).
In case of Hub or DBM exception, `MonitoringHub` cleans up all monitoring processes.

2. Change Hub and DBM processes to daemon.

3. Change multiple threads in DBM to daemon.

PS: it passed pytest with monitoring enabled locally.

### CODE REVIEW

File: parsl/monitoring/monitoring.py [hunk 1]
```diff
@@ -433,10 +448,17 @@ def start(self, priority_msgs, node_msgs, resource_msgs):
...
+def hub_starter(comm_q, exception_q, priority_msgs, node_msgs, resource_msgs, *args, **kwargs):
     hub = Hub(*args, **kwargs)
     comm_q.put((hub.hub_port, hub.ic_port))
-    hub.start(priority_msgs, node_msgs, resource_msgs)
+    hub.logger.info("Starting Hub in Hub starter")
+    try:
+        hub.start(priority_msgs, node_msgs, resource_msgs)
+    except Exception:
+        hub.logger.exception("hub.start exception")
+        exception_q.put("Hub got exception")
```
Comment (yadudoc) [inline parsl/monitoring/monitoring.py]: Could we add exception message itself so that the main process can report it better ?

Comment (ZhuozhaoLi) [reply parsl/monitoring/monitoring.py]: Fixed this.

File: parsl/monitoring/db_manager.py [hunk 2]
```diff
@@ -471,8 +471,8 @@ def dbm_starter(exception_q, priority_msgs, node_msgs, resource_msgs, *args, **k
     dbm.logger.info("Starting dbm in dbm starter")
     try:
         dbm.start(priority_msgs, node_msgs, resource_msgs)
-    except Exception:
+    except Exception as e:
         dbm.logger.exception("dbm.start exception")
-        exception_q.put("DBM got exception")
+        exception_q.put(("DBM", e))
```
Comment (yadudoc) [inline parsl/monitoring/db_manager.py]: Sending the exception object like expects serializability, which I worry may not always work. Maybe stringify the exception and traceback ?

File: parsl/monitoring/monitoring.py [hunk 3]
```diff
@@ -454,9 +456,9 @@ def hub_starter(comm_q, exception_q, priority_msgs, node_msgs, resource_msgs, *a
     hub.logger.info("Starting Hub in Hub starter")
     try:
         hub.start(priority_msgs, node_msgs, resource_msgs)
-    except Exception:
+    except Exception as e:
         hub.logger.exception("hub.start exception")
-        exception_q.put("Hub got exception")
+        exception_q.put(('Hub', e))
```
Comment (yadudoc) [inline parsl/monitoring/monitoring.py]: Same as above

File: parsl/monitoring/db_manager.py (hunk 2 above)
Comment (ZhuozhaoLi) [reply parsl/monitoring/db_manager.py]: fixed

File: parsl/monitoring/monitoring.py (hunk 3 above)
Comment (ZhuozhaoLi) [reply parsl/monitoring/monitoring.py]: fixed

**Pull request merged**